cloudflareupdater.py

```diff
@@ -0,0 +1,206 @@
+import boto3
+import json
+import os
+import urllib3
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(os.environ.get('LOG_LEVEL', logging.DEBUG))
+
+for handler in logger.handlers:
+    handler.setFormatter(logging.Formatter(
+        '%(asctime)s [%(levelname)s](%(name)s) %(message)s'))
+
+for lib_logger in ['botocore', 'boto3', 'urllib3']:
+    logging.getLogger(lib_logger).setLevel(
+        os.environ.get('LIBRARY_LOG_LEVEL', logging.ERROR))
+
+ec2 = boto3.client('ec2')
+
+def get_cloudflare_ip_list():
+    """ Call the CloudFlare API and return a list of IPs """
+    http = urllib3.PoolManager()
+    response = http.request('GET', 'https://api.cloudflare.com/client/v4/ips')
+    payload = json.loads(response.data.decode('utf-8'))
+    logger.info("Retrieved current CloudFlare IPs: %s" % (payload))
+
+    if 'result' in payload:
+        return payload['result']
+
+    raise Exception("Cloudflare response error")
+
+
+def get_aws_security_groups(tag_key, tag_value):
+    """ Return security groups based on `tag_key=tag_value """
+
+    response = ec2.describe_security_groups(Filters=[
+        {
+            'Name': 'tag:{}'.format(tag_key),
+            'Values': [
+                tag_value,
+            ]
+        }])
+
+    return response['SecurityGroups']
+
+
+def check_ipv4_rule_exists(rules, address, port):
+    """ Check if the rule currently exists """
+    logger.debug("Looking for %s and port %i" % (address, port))
+    for rule in rules:
+        for ip_range in rule['IpRanges']:
+            if ip_range['CidrIp'] == address and rule['FromPort'] == port:
+                return True
+
+    return False
+
+
+def add_ipv4_rule(group, address, port):
+    """ Add the IP address/port to the security group """
+    ec2.authorize_security_group_ingress(
```

Comment:
You should probably be using a `try` block here. If the rules are full or anything like that you're going to end up with an eror

Reply:
yeah, but what am I going to do with that error? In its current status it will float to the logs, which seems like the best outcome here..